tests/test_run_reference_vmm.py

```diff
@@ -139,6 +139,14 @@ def shutdown(vmm_process):
     vmm_process.wait(timeout=3)
 
 
+def shutdown_piped(vmm_process, vmm_input_fd):
```

Comment:
At this point I don't think it's feasible to continue to use `vmm_process` as we had before. It becomes more apparent that we need to define a `Vmm` class that keeps the `stdin` and `stdout` configuration, along with the other configurations that are needed in order to not duplicate the `shutdown` and the `expect_string` functions.

At a minimum, if we do not implement this in this PR, we should have a generic shutdown function that receives as a parameter the file descriptor, and then write a variant for it where the fd is actually stdin. Same approach should be used for `expect_string`